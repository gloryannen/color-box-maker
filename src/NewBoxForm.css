.newBoxFormMain {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  text-align: center;
}

form {
  width: 30%;
  background-color: rgb(255, 231, 200);
  margin: 2em;
  font-family: Verdana, Geneva, Tahoma, sans-serif;
}
input {
  width: 80%;
  border: 0;
  border-bottom: 1px solid gray;
  font-size: 14px;
  font-weight: 300;
  color: #3a3a3a;
  letter-spacing: 5px;
  background-color: rgb(255, 231, 200);
  margin: 8px;
}

input:focus {
  outline: 0;
  border-bottom: 1px solid rgb(27, 27, 27);
}

button {
  cursor: pointer;
  border: none;
  color: rgb(29, 29, 29);
  margin: 10px 10px;
  padding: 8px 25px;
  font-weight: bold;
  font-size: 15px;
  background-color: #f6f6f0a4;
  letter-spacing: 2px;
}
