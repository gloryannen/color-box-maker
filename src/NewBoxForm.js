import React, { useState } from 'react';
import { v4 as uuid } from 'uuid';
import './NewBoxForm.css';

const NewBoxForm = ({ addBox }) => {
  // Set a form inputs to empty strings/values
  const INITIAL_STATE = {
    backgroundColor: '',
    width: '',
    height: ''
  };

  const [formData, setFormData] = useState(INITIAL_STATE);
  // Handle input values
  const handleChange = (e) => {
    const { name, value } = e.target;
    setFormData(formData => ({
      ...formData,
      [name]: value
    }));
  }

  // Handle form submission - create new box with params given, assign id, clear inputs
  const handleSubmit = (e) => {
    e.preventDefault();
    addBox({ ...formData, id: uuid() });
    setFormData(INITIAL_STATE)
  };

  //Color box form
  return (
    <div className='newBoxFormMain'>
      <form onSubmit={ handleSubmit }>
        <h2>Color Box Maker</h2>
        <input
          id='backgroundColor'
          type='text'
          name='backgroundColor'
          value={ formData.backgroundColor }
          onChange={ handleChange }
          placeholder='Color'
          required
        />
        <input
          id='width'
          type='text'
          name='width'
          value={ formData.width }
          onChange={ handleChange }
          placeholder='Width(px)'
          required
        />
        <input
          id='height'
          type='text'
          name='height'
          value={ formData.height }
          onChange={ handleChange }
          placeholder='Height(px)'
          required
        />
        <button>Create Box</button>
      </form>
    </div>
  )
}

export default NewBoxForm;