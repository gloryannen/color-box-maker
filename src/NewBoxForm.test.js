import React from 'react';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import NewBoxForm from './NewBoxForm.js';

// Smoke test
it("renders without crashing", function () {
  render(<NewBoxForm />);
});

// Snapshot test
it("matches snapshot", function () {
  const { asFragment } = render(<NewBoxForm />);
  expect(asFragment()).toMatchSnapshot();
});
