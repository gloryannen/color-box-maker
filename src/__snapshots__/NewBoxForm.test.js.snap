// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="newBoxFormMain"
  >
    <form>
      <h2>
        Color Box Maker
      </h2>
      <input
        id="backgroundColor"
        name="backgroundColor"
        placeholder="Color"
        required=""
        type="text"
        value=""
      />
      <input
        id="width"
        name="width"
        placeholder="Width(px)"
        required=""
        type="text"
        value=""
      />
      <input
        id="height"
        name="height"
        placeholder="Height(px)"
        required=""
        type="text"
        value=""
      />
      <button>
        Create Box
      </button>
    </form>
  </div>
</DocumentFragment>
`;
